File: browse_stock.py

#------------------------------------#
#  SISTEMA GESTAO ONLINE API
#  adquirindo estoque atual
#  autor: softupsistemas
#-------------------------------------#


#importa modulo sys
import sys

# importando a biblioteca de requisições
import requests 

# importando a biblioteca de JSON
import json 
  
# definindo url da requisicao
url_request = "https://www.sistemagestaoonline.com.br/sistema/rest/api_sgo.wsconsultaestoque"
  
# definindo produto referencia 
produtoReferencia = "123"

# montando header
header = {'Content-type': 'application/json'}

# definindo chave de acesso 
chaveAcesso = "et53r4g934j90tj39gjhy93jy39jg93=gb"
  
# montando body da requisição 
data = {'Chave':chaveAcesso, 
        'ProdutoReferencia':produtoReferencia} 
  
# fazendo requisição POST
r = requests.post(url = url_request, data = data, headers = header) 
  
# extracting data in json format 
data = r.json() 

# printa resposta
print(data)

File: browse_stock.js

//inicializa instancia de XMLHtppRequest
var xhr = new XMLHttpRequest();

//define url da requisição
var url_request = "https://www.sistemagestaoonline.com.br/sistema/rest/api_sgo.wsconsultaestoque"

//monta requisição POST sincrona
xhr.open('POST', url_request, false);

//define body
var chave = "et53r4g934j90tj39gjhy93jy39jg93=gb"
var produtoReferencia = "123"

var params = JSON.stringify({ "Chave": chave,"ProdutoReferencia":produtoReferencia });

//define header
http.setRequestHeader("Content-type", "application/json");

//executa requisição
xhr.send(params);
 
//interpreta e printa resultado
xhr.onreadystatechange = processRequest;
 
function processRequest(e) {
    
    if (xhr.readyState == 4 && xhr.status == 200) {
         var response = JSON.parse(xhr.responseText);
        alert(response.EstoqueAtual);
    } 
}